### DISCUSSION TITLE
What am I doing wrong here with trying to create an override for an assignment?

### DISCUSSION THREAD

**Discussion by lexicalmichaellake** [ucfopen/canvasapi#534] [Q&A] [answered]
Hi,

First of all, I LOVE THIS PACKAGE! It's already saved me a tonne of time. You all are amazing people! 

EDIT: many thanks to the combined powers of @justin0022 and @Lannro and the chance to use their reference code over at [https://github.com/ubccapico](https://github.com/ubccapico)!

If you're getting a  "create_override() takes 1 positional argument but 2 were given" error. when trying to make an override, make sure your create_override command looks at least something like this 
```
course = canvas.get_course(######)
assignment = course.get_assignment(######)

d = datetime(2022, 10, 30, 11, 59)

assignment.create_override(assignment_override={"student_ids": [###], "title": "assignment_x_1_etc", "due_at": d})
```

Thank you all for your work here and for being here!

**Comment by bennettscience** [comment, marked as answer]
Glad you got it worked out. Most `POST` requests in the library require your data to be passed in as a `kwarg` because that's what each endpoint expects. Wrapping it in the `kwarg` name is usually the solution to these kinds of errors.
